### ISSUE TITLE
 Type 'typeof GlobalComponent' is not assignable to type 'ComponentConstructor'

### ISSUE THREAD

**Issue by alexanderkoller** [jovotech/jovo-cli#286]
## I'm submitting a...

- [x] Bug report
- [ ] Feature request
- [ ] Documentation issue or request
- [ ] Other... Please describe:

## Expected Behavior
After creating a fresh Jovo project with `jovo new xxx` and selecting the default configuration, `jovo run` should start a Jovo instance.

## Current Behavior
`jovo run` prints an error and does not start the local server. The full log is below.

This is really strange, because I already created a Jovo project yesterday, and that worked fine. Is it possible that Node or npm has some global state somewhere that interferes with the new project? Sorry is this is a newbie question, I don't usually work with Javascript.

```
turing:tmp koller$ jovo new bug
                                                                                
jovo new: Create a new Jovo project
  >> Learn more: https://jovo.tech/docs/cli/new
                                                                                
🔮 Welcome to the Jovo CLI Wizard
                                                                                
✔ Pick a preset: › default
                                                                                
🔧 Setting everything up
                                                                                
✔ Creating new directory bug/
  /private/tmp/bug
✔ Downloading and extracting template
✔ Generating package.json
✔ Installing npm dependencies
                                                                                
🎉 Successfully created your project!
                                                                                
turing:tmp koller$ cd bug
turing:bug koller$ jovo run
                                                                                
jovo run: Start the local development server and test your app using the Jovo Debugger
  >> Learn more: https://jovo.tech/docs/cli/run


> jovo-v4-template@4.0.0 start:dev
> tsc-watch --onSuccess "node ./dist/app.dev.js --jovo-webhook" --noClear



----------------------
9:33:11 AM - Starting compilation in watch mode...

src/app.ts(25,16): error TS2322: Type 'typeof GlobalComponent' is not assignable to type 'ComponentConstructor<BaseComponent<ComponentData, UnknownObject>> | ComponentDeclaration<ComponentConstructor<BaseComponent<ComponentData, UnknownObject>>>'.
  Type 'typeof GlobalComponent' is not assignable to type 'ComponentConstructor<BaseComponent<ComponentData, UnknownObject>>'.
src/app.ts(25,33): error TS2322: Type 'typeof LoveHatePizzaComponent' is not assignable to type 'ComponentConstructor<BaseComponent<ComponentData, UnknownObject>> | ComponentDeclaration<ComponentConstructor<BaseComponent<ComponentData, UnknownObject>>>'.
  Type 'typeof LoveHatePizzaComponent' is not assignable to type 'ComponentConstructor<BaseComponent<ComponentData, UnknownObject>>'.
src/components/GlobalComponent.ts(14,2): error TS2345: Argument of type 'typeof GlobalComponent' is not assignable to parameter of type 'ComponentConstructor<BaseComponent<ComponentData, UnknownObject>>'.
  Types of construct signatures are incompatible.
    Type 'new (jovo: Jovo<JovoRequest, JovoResponse, any, JovoUser<any>, JovoDevice<any, string>, any>, options?: ComponentOptions<UnknownObject> | undefined) => GlobalComponent' is not assignable to type 'new (jovo: Jovo<JovoRequest, JovoResponse, any, JovoUser<any>, JovoDevice<any, string>, any>, options?: ComponentOptionsOf<BaseComponent<ComponentData, UnknownObject>> | undefined) => BaseComponent<...>'.
      Types of parameters 'options' and 'options' are incompatible.
        Type 'ComponentOptionsOf<BaseComponent<ComponentData, UnknownObject>> | undefined' is not assignable to type 'ComponentOptions<UnknownObject> | undefined'.
src/components/GlobalComponent.ts(17,27): error TS2769: No overload matches this call.
  Overload 1 of 2, '(constructor: ComponentConstructor<LoveHatePizzaComponent>, handler?: "lovesPizza" | "hatesPizza" | "START" | "UNHANDLED" | undefined): Promise<...>', gave the following error.
    Argument of type 'typeof LoveHatePizzaComponent' is not assignable to parameter of type 'ComponentConstructor<LoveHatePizzaComponent>'.
      Types of construct signatures are incompatible.
        Type 'new (jovo: Jovo<JovoRequest, JovoResponse, any, JovoUser<any>, JovoDevice<any, string>, any>, options?: ComponentOptions<UnknownObject> | undefined) => LoveHatePizzaComponent' is not assignable to type 'new (jovo: Jovo<JovoRequest, JovoResponse, any, JovoUser<any>, JovoDevice<any, string>, any>, options?: ComponentOptionsOf<LoveHatePizzaComponent> | undefined) => LoveHatePizzaComponent'.
          Types of parameters 'options' and 'options' are incompatible.
            Type 'ComponentOptionsOf<LoveHatePizzaComponent> | undefined' is not assignable to type 'ComponentOptions<UnknownObject> | undefined'.
              Type 'ComponentOptionsOf<LoveHatePizzaComponent>' is not assignable to type 'ComponentOptions<UnknownObject>'.
  Overload 2 of 2, '(constructorOrName: string | ComponentConstructor<BaseComponent<ComponentData, UnknownObject>>, handler?: string | undefined): Promise<...>', gave the following error.
    Argument of type 'typeof LoveHatePizzaComponent' is not assignable to parameter of type 'string | ComponentConstructor<BaseComponent<ComponentData, UnknownObject>>'.
      Type 'typeof LoveHatePizzaComponent' is not assignable to type 'ComponentConstructor<BaseComponent<ComponentData, UnknownObject>>'.
src/components/LoveHatePizzaComponent.ts(14,2): error TS2345: Argument of type 'typeof LoveHatePizzaComponent' is not assignable to parameter of type 'ComponentConstructor<BaseComponent<ComponentData, UnknownObject>>'.
  Types of construct signatures are incompatible.
    Type 'new (jovo: Jovo<JovoRequest, JovoResponse, any, JovoUser<any>, JovoDevice<any, string>, any>, options?: ComponentOptions<UnknownObject> | undefined) => LoveHatePizzaComponent' is not assignable to type 'new (jovo: Jovo<JovoRequest, JovoResponse, any, JovoUser<any>, JovoDevice<any, string>, any>, options?: ComponentOptionsOf<BaseComponent<ComponentData, UnknownObject>> | undefined) => BaseComponent<...>'.
      Types of parameters 'options' and 'options' are incompatible.
        Type 'ComponentOptionsOf<BaseComponent<ComponentData, UnknownObject>> | undefined' is not assignable to type 'ComponentOptions<UnknownObject> | undefined'.
          Type 'ComponentOptionsOf<BaseComponent<ComponentData, UnknownObject>>' is not assignable to type 'ComponentOptions<UnknownObject>'.
            Type 'UnknownObject | undefined' is not assignable to type 'UnknownObject'.
              Type 'undefined' is not assignable to type 'UnknownObject'.

9:33:14 AM - Found 5 errors. Watching for file changes.
```

## Your Environment

```
turing:bug koller$ jovo --version
                                                                                
@jovotech/cli: 4.0.3 

Jovo packages of the current project :
  - @jovotech/cli-command-build: 4.0.1
  - @jovotech/cli-command-deploy: 4.0.1
  - @jovotech/cli-command-get: 4.0.1
  - @jovotech/cli-command-new: 4.0.1
  - @jovotech/cli-command-run: 4.0.1
  - @jovotech/cli-core: 4.0.1
  - @jovotech/common: 4.0.2
  - @jovotech/db-filedb: 4.1.1
  - @jovotech/filebuilder: 0.0.1
  - @jovotech/framework: 4.1.1
  - @jovotech/model: 4.0.0
  - @jovotech/model-nlpjs: 4.0.0
  - @jovotech/nlu-nlpjs: 4.1.1
  - @jovotech/output: 4.0.2
  - @jovotech/platform-core: 4.1.1
  - @jovotech/plugin-debugger: 4.1.1
  - @jovotech/server-express: 4.1.1
                                                                                
Environment:
  System:
    OS: macOS 11.4
  Binaries:
    Node: 14.15.2 - /usr/local/bin/node
    npm: 8.3.1 - /usr/local/bin/npm


turing:bug koller$ npm list|grep jovo
jovo-v4-template@4.0.0 /private/tmp/bug
├── @jovotech/cli-command-build@4.0.1
├── @jovotech/cli-command-deploy@4.0.1
├── @jovotech/cli-command-get@4.0.1
├── @jovotech/cli-command-new@4.0.1
├── @jovotech/cli-command-run@4.0.1
├── @jovotech/cli-core@4.0.1
├── @jovotech/db-filedb@4.1.1
├── @jovotech/filebuilder@0.0.1
├── @jovotech/framework@4.1.1
├── @jovotech/plugin-debugger@4.1.1
├── @jovotech/server-express@4.1.1
```

**Comment by aswetlow** [comment]
Hey @alexanderkoller 

Yes, this is weird. We published a new version 2 days ago and it worked. Now I get the same error. 

We're investigating.

**Comment by m-ripper** [comment]
Hello @alexanderkoller,

after some checking I found the bug. It seems that the latest update of [type-fest](https://github.com/sindresorhus/type-fest) had an update just a few hours ago.

This update seems to have introduced breaking changes for us. 

As a workaround until we publish a fix, you can add `"type-fest": "2.9.0"` to your `package.json` then reinstall and it should work.

Related PR: [jovo-framework#1191](https://github.com/jovotech/jovo-framework/pull/1191/files)

**Comment by alexanderkoller** [comment]
That's great, thanks a lot!
